csrc/dynamic_transform.cpp

```diff
@@ -98,25 +92,17 @@ class DynamicTransformInitialInfoBuilder : public IterVisitor {
     auto out_tv = op->out()->as<TensorView>();
     // If there's no symbolic axis, this is a static reshape op
     if (out_tv->domain()->hasSymbolicAxis()) {
-      info_.dynamic_reshapes_.push_back(op);
+      info_.dynamic_reshapes_.push_back(out_tv);
 
       // Input and output extent expressions both affect concretization
       const auto& inp_dom =
           TensorDomain::noReductions(inp_tv->getMaybeRFactorDomain());
       for (const auto id : inp_dom) {
-        // Try and evaluate the extent so that intermediate expressions are
-        // cached in expr_eval_
-        auto ext = info_.expr_eval_.evaluate(id->extent());
-        if (!ext.has_value()) {
-          leaf_dynamic_vals_.push_back(id->extent());
-        }
+        leaf_dynamic_vals_.push_back(id->extent());
```

Comment:
I think intermediate expressions are not cached in the ExpressionEvaluator unless we use a `PrecomputedValues`. I tried to get that to work also, and while I think it may be possible, it would preclude rebinding variables when we see different inputs since `PrecomputedValues::bindInputs()` checks whether each `Val` has previously been bound, while `ExpressionEvaluator::bind()` does not.